File: python/di-string-match.py
class Solution:
    def diStringMatch(self, s: str) -> List[int]:
        n = len(s)
        minimal = 0
        maximal = n
        output = []
        for c in s:
            if c == "I":
                output.append(minimal)
                minimal += 1
            if c == "D":
                output.append(maximal)
                maximal -= 1
        output.append(minimal)
        return output

File: rust/di-string-match.rs
impl Solution {
    pub fn di_string_match(s: String) -> Vec<i32> {
        let mut output = Vec::new();
        let mut min = 0;
        let mut max = s.len() as i32;
        for c in s.chars() {
            if c == 'I' {
                output.push(min);
                min += 1;
            } else {
                output.push(max);
                max -= 1;
            }
        }
        output.push(min);
        return output;
    }
}
